tests/helpers_tests/test_validator.py

```diff
@@ -168,6 +171,55 @@ def test_is_attribute_valid(self):
     self.assertTrue(validator.is_attribute_valid('test_attribute', ""))
     self.assertTrue(validator.is_attribute_valid('test_attribute', 'test_value'))
 
+    # test if attribute value is a number, it calls is_finite_number and returns it's result
+    with mock.patch('optimizely.helpers.validator.is_finite_number',
+                    return_value=True) as is_finite:
+      self.assertTrue(validator.is_attribute_valid('test_attribute', 5))
+
+    is_finite.assert_called_once_with(5)
+
+    with mock.patch('optimizely.helpers.validator.is_finite_number',
+                    return_value=False) as is_finite:
+      self.assertFalse(validator.is_attribute_valid('test_attribute', 5.5))
+
+    is_finite.assert_called_once_with(5.5)
+
+    with mock.patch('optimizely.helpers.validator.is_finite_number',
+                    return_value='abc') as is_finite:
+      self.assertEqual('abc', validator.is_attribute_valid('test_attribute', 0))
+
+    is_finite.assert_called_once_with(0)
+
+  def test_is_finite_number(self):
+    """ Test that it returns true if value is a number and is not more than 1e53 if an Integer,
+    and not one of NAN, INF or -INF if it's a double. """
+
+    # test non number values
+    self.assertFalse(validator.is_finite_number('HelloWorld'))
+    self.assertFalse(validator.is_finite_number(True))
+    self.assertFalse(validator.is_finite_number(False))
+    self.assertFalse(validator.is_finite_number(None))
+    self.assertFalse(validator.is_finite_number({}))
+    self.assertFalse(validator.is_finite_number([]))
+    self.assertFalse(validator.is_finite_number(()))
+
+    # test invalid numbers
+    self.assertFalse(validator.is_finite_number((float('inf'))))
+    self.assertFalse(validator.is_finite_number((float('-inf'))))
+    self.assertFalse(validator.is_finite_number((float('nan'))))
+    self.assertFalse(validator.is_finite_number(int(1e53) + 1))
+    if PY2:
+      self.assertFalse(validator.is_finite_number(long(1e53) + 1))
```

Comment:
As implied by my comments in the code under test, we also want assertions like these:
```python
self.assertFalse(validator.is_finite_number(float(1e53) + 1.0))
self.assertFalse(validator.is_finite_number(-int(1e53) - 1))
if PY2:
  self.assertFalse(validator.is_finite_number(-long(1e53) - 1))
self.assertFalse(validator.is_finite_number(-float(1e53) - 1.0))
```
(maybe reordered as you deem fit)